sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -496,6 +505,233 @@ def reader(self, test_bigquery_client=None):
+      :data:`None` if the table reference is specified entirely by the table
+      argument.
+    project (str): The ID of the project containing this table.
+    query (str): A query to be used instead of arguments table, dataset, and
+      project.
+      validate (bool): If :data:`True`, various checks will be done when source
+      gets initialized (e.g., is table present?). This should be
+      :data:`True` for most scenarios in order to catch errors as early as
+      possible (pipeline construction instead of pipeline execution). It
+      should be :data:`False` if the table is created during pipeline
+      execution by a previous step.
+    coder (~apache_beam.coders.coders.Coder): The coder for the table
+      rows. If :data:`None`, then the default coder is
+      :class:`~apache_beam.io.gcp.bigquery._BigQueryRowCoder`,
+      which will interpret every line in a file as a JSON serialized
+      dictionary. This argument needs a value only in special cases when
+      returning table rows as dictionaries is not desirable.
+    use_standard_sql (bool): Specifies whether to use BigQuery's standard SQL
+      dialect for this query. The default value is :data:`False`.
+      If set to :data:`True`, the query will use BigQuery's updated SQL
+      dialect with improved standards compliance.
+      This parameter is ignored for table inputs.
+    flatten_results (bool): Flattens all nested and repeated fields in the
+      query results. The default value is :data:`True`.
+    kms_key (str): Experimental. Optional Cloud KMS key name for use when
+      creating new tables.
+    gcs_bucket_name (str): The name of the Google Cloud Storage bucket where
+      the extracted table should be written.
+   """
+  def __init__(self, table=None, dataset=None, project=None, query=None,
+               validate=False, coder=None, use_standard_sql=False,
+               flatten_results=True, kms_key=None, gcs_bucket_name=None):
+    if gcs_bucket_name is None:
+      raise ValueError('The name of the GCS bucket must be specified')
+    self.gcs_bucket_name = gcs_bucket_name
+
+    if table is not None and query is not None:
+      raise ValueError('Both a BigQuery table and a query were specified.'
+                       ' Please specify only one of these.')
+    elif table is None and query is None:
+      raise ValueError('A BigQuery table or a query must be specified')
+    elif table is not None:
+      self.table_reference = bigquery_tools.parse_table_reference(
+          table, dataset, project)
+      self.query = None
+      self.use_legacy_sql = True
+    else:
+      self.query = query
+      # TODO(BEAM-1082): Change the internal flag to be standard_sql
+      self.use_legacy_sql = not use_standard_sql
+      self.table_reference = None
+
+    self.project = project
+    self.validate = validate
+    self.flatten_results = flatten_results
+    self.coder = coder or _BigQueryRowCoder
+    self.kms_key = kms_key
```

Comment:
Ah nvm I see now. This kms key should be used when creating a temporary dataset with the query, right? It seems likeit's not used for that atm.

Reply:
Thanks for spotting this. I will write a fix so that the kms key will be used when creating a temporary dataset. There are no other places where kms key should be used, right?